crates/ruff/src/rules/isort/helpers.rs

```diff
@@ -102,103 +102,148 @@ fn match_docstring_end(body: &[Stmt]) -> Option<Location> {
     Some(stmt.end_location.unwrap())
 }
 
-/// Find the end of the first token that isn't a docstring, comment, or
-/// whitespace.
-pub fn find_splice_location(body: &[Stmt], locator: &Locator) -> Location {
-    // Find the first AST node that isn't a docstring.
-    let mut splice = match_docstring_end(body).unwrap_or_default();
+/// Find the location at which a "top-of-file" import should be inserted,
+/// along with a prefix and suffix to use for the insertion.
+///
+/// For example, given the following code:
+///
+/// ```python
+/// """Hello, world!"""
+///
+/// import os
+/// ```
+///
+/// The location returned will be the start of the `import os` statement,
+/// along with a trailing newline suffix.
+pub fn top_of_file_insertion(
```

Comment:
Nit: `pub(crate)` or make it even private (specific to `isort`). we should avoid making things public because Rust's dead code analysis can only flag non-pub types as unused.